src/utils/workspaceUtils.ts

```diff
@@ -3,10 +3,47 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { workspace } from "vscode";
+import * as path from 'path';
+import { OpenDialogOptions, workspace, WorkspaceFolder } from "vscode";
+import { IAzureQuickPickItem } from "vscode-azureextensionui";
+import { ext } from '../extensionVariables';
+import { localize } from "./localize";
 
 // tslint:disable-next-line:export-name
 export function getSingleRootFsPath(): string | undefined {
     // if this is no workspace or a multi-root workspace, return undefined
     return workspace.workspaceFolders && workspace.workspaceFolders.length === 1 ? workspace.workspaceFolders[0].uri.fsPath : undefined;
 }
+
+export async function selectWorkspaceFolder(placeHolder: string, getSubPath?: (f: WorkspaceFolder) => string | undefined | Promise<string | undefined>): Promise<string> {
+    return await selectWorkspaceItem(
+        placeHolder,
+        {
+            canSelectFiles: false,
+            canSelectFolders: true,
+            canSelectMany: false,
+            defaultUri: workspace.workspaceFolders && workspace.workspaceFolders.length > 0 ? workspace.workspaceFolders[0].uri : undefined,
+            openLabel: localize('select', 'Select')
+        },
+        getSubPath);
+}
+
+export async function selectWorkspaceItem(placeHolder: string, options: OpenDialogOptions, getSubPath?: (f: WorkspaceFolder) => string | undefined | Promise<string | undefined>): Promise<string> {
+    let folder: IAzureQuickPickItem<string | undefined> | undefined;
+    if (workspace.workspaceFolders) {
```

Comment:
The scenario where I have no folders open was particularly jarring for me. Maybe for static web apps we need to show a quick pick even if "Browse..." is the only option - just to make sure users see the placeholder